### PR TITLE
New option `onlyEnums` for Typescript

### PR DESCRIPTION
**Pull request by Borduhh**
🚨 **IMPORTANT: Please do not create a Pull Request without creating an issue first.**

_Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request._

## Description

Added so that typescript projects are able to export types and enums independently. This is especially useful for using generated types as a namespace in monorepos.

Related #3475 (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots/Sandbox (if appropriate/relevant):

Adding links to sandbox or providing screenshots can help us understand more about this PR and take action on it as appropriate

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Environment**:

- OS:
- `@graphql-codegen/...`:
- NodeJS:

## Checklist:

- [x] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Borduhh** [comment]
Are any changes needed for this? We're going to be using this in one of our projects ASAP after being merged.

**Comment by Borduhh** [comment]
@charlypoly any ETA on a release date for this? We're really looking forward to implementing this in our project.

**Comment by DrChrispoper** [comment]
This would be a great addition. Thanks for the contribution @Borduhh

File: dev-test/githunt/types.preResolveTypes.onlyOperationTypes.ts [hunk 1]
```diff
@@ -12,6 +12,50 @@ export type Scalars = {
   Float: number;
 };
 
+/** A comment about an entry, submitted by a user */
```
Comment (charlypoly) [inline dev-test/githunt/types.preResolveTypes.onlyOperationTypes.ts:15]: @Borduhh this output changing looks like a regression on the `onlyOperationTypes` option 👀

**Review by charlypoly — CHANGES_REQUESTED** [review]

File: dev-test/githunt/types.preResolveTypes.onlyOperationTypes.ts (hunk 1 above)
Comment (Borduhh) [reply dev-test/githunt/types.preResolveTypes.onlyOperationTypes.ts]: That would be. Found the typo and fixed it 🎉. There should be not regression now.

**Review by charlypoly — APPROVED** [review]

**Comment by charlypoly** [comment]
Released in `@graphql-codegen/typescript@2.4.10` and related packages (see https://github.com/dotansimha/graphql-code-generator/pull/7822)

**Pull request merged**